### PR TITLE
OPHAKTKEH-133: informal emails sending endpoint

### PR DESCRIPTION
**Pull request by mikhuttu**
Endpoint for sending informal emails to selected translators. Tested this locally by commenting out the authorization via role VIRKAILIJA under WebSecurityConfig.

Implemented on top of https://github.com/Opetushallitus/akt/pull/70/files because ClerkTranslatorService seemed like a suitable place for putting the service level logic.

### CODE REVIEW

File: src/main/resources/db/changelog/db.changelog-1.0.xml [hunk 1]
```diff
@@ -407,4 +407,13 @@
         </addColumn>
     </changeSet>
 
+    <changeSet id="2022-01-04-add-new-email-types" author="mikhuttu">
+        <insert tableName="email_type">
+            <column name="name" value="AUTHORISATION_EXPIRY"/>
```
Comment (mikhuttu) [inline src/main/resources/db/changelog/db.changelog-1.0.xml:416]: Insertion of AUTHORISATION_EXPIRY could be removed from this pull request but then yet another changeSet has to be added in OPHAKTKEH-132. That might be useful though in case someone wants to work on that while having this PR still in review phase (and not on top of this one).

File: src/main/java/fi/oph/akt/service/ClerkTranslatorService.java [hunk 2]
```diff
@@ -244,4 +252,25 @@ private LanguagePairsDictDTO getLanguagePairsDictDTO() {
 		return translatorDetails.stream().map(TranslatorDetails::town).distinct().sorted().toList();
 	}
 
+	public void createInformalEmails(InformalEmailRequestDTO emailRequestDTO) {
```
Comment (terova) [inline src/main/java/fi/oph/akt/service/ClerkTranslatorService.java]: Mutu sanoo että virkailijan sähköpostien lähetys kuuluisi omaan serviceen.

File: src/main/java/fi/oph/akt/service/ClerkTranslatorService.java [hunk 3]
```diff
@@ -244,4 +252,25 @@ private LanguagePairsDictDTO getLanguagePairsDictDTO() {
...
+
+		if (translators.size() != distinctTranslatorIds.size()) {
+			throw new IllegalArgumentException("Each translator by provided translatorIds not found");
+		}
+
+		translators.forEach(translator -> {
+			// @formatter:off
+			EmailData emailData = EmailData.builder()
+					.sender("AKT")
+					.recipient("translator" + translator.getId() + "@test.fi")
```
Comment (terova) [inline src/main/java/fi/oph/akt/service/ClerkTranslatorService.java]: Tänne TODO kuten ContactRequestService:ssa.

File: src/test/java/fi/oph/akt/service/ClerkTranslatorServiceTest.java [hunk 4]
```diff
@@ -358,4 +400,74 @@ public void listShouldReturnProperDataForTranslatorWithMultipleAuthorisationTerm
...
+		entityManager.persist(authorisationTerm);
+
+		Long tId = translatorRepository.findAll().get(0).getId();
+
+		InformalEmailRequestDTO emailRequestDTO = InformalEmailRequestDTO.builder().translatorIds(List.of(tId, tId))
+				.subject("otsikko").body("viesti").build();
+
+		clerkTranslatorService.createInformalEmails(emailRequestDTO);
+
+		verify(emailService, times(1)).saveEmail(any(), emailDataCaptor.capture());
```
Comment (terova) [inline src/test/java/fi/oph/akt/service/ClerkTranslatorServiceTest.java]: times(1) on verify() default.

**Review by terova — COMMENTED** [review]
Nopeasti katsottuna ihan hyvä.

**Review by terova — APPROVED** [review]
